### PR TITLE
Model to json

### PR DESCRIPTION
**Pull request by mdip226**

### CODE REVIEW

**Comment by ethangreen-dev** [comment]
#489 implemented in commit e225b701085817fcd76e00f223c7d6f1964d936a.

**Comment by ethangreen-dev** [comment]
A couple issues remain before I can feel confident merging this branch:

- Variables that represent the object need to be removed. Namely, `_hash` and `_translation_table`.
- The translation table needs some sort of "blocklist" so we're not anonymizing variable names. The implementation within Jsonify shouldn't be complex, but the keys to ignore need to be generated from the Model.
- Add JSON coding support to `type`. This is to ensure that `model.run()` args can be coded while retaining the original type information.

As always, more tests need to be written. For instance, `model.run()` == `model.to_anon().run().from_anon()`. 

I'm considering yanking Jsonify out into a package so it can easily be used in SpacialPy. What are your thoughts on this?

**Comment by ethangreen-dev** [comment]
- [X] Variables that represent the object need to be removed. Namely, _hash and 

- [X] The translation table needs some sort of "blocklist" so we're not anonymizing variable names. The implementation within Jsonify shouldn't be complex, but the keys to ignore need to be generated from the 

- [X] Add JSON coding support to type. This is to ensure that model.run() args can be coded while retaining the original type information.

As always, `model_to_json` requires more unit tests. I'll be adding these in the next couple days as I integrate these changes into stochss-remote.

File: gillespy2/core/functiondefinition.py [hunk 1]
```diff
@@ -35,7 +38,7 @@ def sanitized_function(self, species_mappings, parameter_mappings):
                        reverse=True)
         replacements = [parameter_mappings[name] if name in parameter_mappings else species_mappings[name]
                         for name in names]
-        sanitized_function = self.function
+        sanitized_function = self.function_string
```
Comment (BryanRumsey) [inline gillespy2/core/functiondefinition.py:41]: `self.function` should be the sanitized function.  `self.function_string` is only used by StochSS

File: gillespy2/core/model.py [hunk 2]
```diff
@@ -202,6 +209,42 @@ def decorate(header):
...
+            dict(zip((str(x.name) for x in assignments), (f"AR_{x + 100}" for x in range(0, len(assignments))))),
+            dict(zip((str(x.name) for x in rates), (f"RR_{x + 100}" for x in range(0, len(rates))))),
+            dict(zip((str(x.name) for x in events), (f"E_{x + 100}" for x in range(0, len(events))))),
+            dict(zip((str(x.name) for x in functions), (f"F_{x + 100}" for x in range(0, len(functions))))),
+        ))
+
+        return TranslationTable(to_anon=translation_table)
+
+    def remote_solver_hash(self):
+        """ Creates an md5 hash of an anonymized version of the model to be used for caching """
```
Comment (BryanRumsey) [inline gillespy2/core/model.py:240]: reminder to remove this function

**Review by BryanRumsey — CHANGES_REQUESTED** [review]

File: gillespy2/core/functiondefinition.py (hunk 1 above)
Comment (ethangreen-dev) [reply gillespy2/core/functiondefinition.py]: Fixed.

File: gillespy2/core/model.py (hunk 2 above)
Comment (ethangreen-dev) [reply gillespy2/core/model.py]: Removed. Thanks for the reminder!

**Review by jtcooper10 — CHANGES_REQUESTED** [review]

**Review by BryanRumsey — APPROVED** [review]

**Review by jtcooper10 — APPROVED** [review]

**Pull request merged**